### PR TITLE
Update DOCKER docs for 8.10 release

### PR DESCRIPTION
**Pull request by jedrazb**
## Closes https://github.com/elastic/enterprise-search-team/issues/5688

Also verified that https://github.com/elastic/enterprise-search-team/issues/5025 is fixed. Couldn't reproduce it so marking it as fixed.

Update Docker instructions.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: docs/DOCKER.md [hunk 1]
```diff
@@ -112,17 +82,16 @@ docker run \
...
-  - If you are using nightly builds, you will need to run `docker pull docker.elastic.co/enterprise-search/elastic-connectors:8.7.0.0-SNAPSHOT` before starting the service. This ensures you're using the latest version of the Docker image.
-- `-c /config/config.yml` - replace `config.yml` with the name of the config file you've put in the directory you've created in step 2. 
-  - Normally, if you run a single instance of the Connector Service you can use `config.yml`. 
-  - If you're planning to run multiple versions of the service, you can put configs for different service versions in a single directory and update this setting to point to different config files.
+- `docker.elastic.co/enterprise-search/elastic-connectors:8.10.0.0-SNAPSHOT` - adjust the version for the connectors to match your Elasticsearch deployment version.
+  - For Elasticsearch of version 8.7 you can use `elastic-connectors:8.10.0.0`for a stable revision of the connectors, or `elastic-connectors:8.10.0.0-SNAPSHOT` if you want the latest nightly build of the connectors (not recommended).
```
Comment (navarone-feekery) [inline docs/DOCKER.md:95]: Versions are mismatched (and possibly another argument for not including exact versions in the docs 😄)
```suggestion
  - For Elasticsearch of version 8.10 you can use `elastic-connectors:8.10.0.0`for a stable revision of the connectors, or `elastic-connectors:8.10.0.0-SNAPSHOT` if you want the latest nightly build of the connectors (not recommended).
```

File: docs/DOCKER.md [hunk 2]
```diff
@@ -41,52 +41,22 @@ curl https://raw.githubusercontent.com/elastic/connectors-python/main/config.yml
...
-  mssql: connectors.sources.mssql:MSSQLDataSource
+# When connecting to your cloud deployment you should edit the host value
+elasticsearch.host: http://host.docker.internal:9200
+
+# Using `api_key` is recommended instead of `username`/`password`
+elasticsearch.api_key: <ELASTICSEARCH_API_KEY>
+# elasticsearch.username: elastic
+# elasticsearch.password: <YOUR_PASSWORD>
+
+connectors:
```
Comment (leemthompo) [inline docs/DOCKER.md:52]: `sidebar`: should these be commented out in the _actual_ config file. Wondering if that's unnecessary 🤔

Comment (jedrazb) [reply docs/DOCKER.md:52]: They are commented out in the actual config file. Let's get rid of them. I added a note at the bottom of the config that you can use username/password as well.

**Review by navarone-feekery — APPROVED** [review]
LGTM from a technical perspective 🚀

**Pull request merged**